src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,252 @@
+   * the value is never updated.
+   */
+  private final Map<Sequence, Long> sequenceExecutionTime = new HashMap<>();
+
+  /**
+   * Bias input selection towards lower-cost sequences. We first compute and update the weights of
+   * all the candidates within the candidate list before making our selection.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    double totalWeight = computeWeightForCandidates(candidates);
+
+    Sequence selectedSequence = Randomness.randomMemberWeighted(candidates, weightMap, totalWeight);
+    CollectionsPlume.incrementMap(sequenceSelectionCount, selectedSequence);
+    return selectedSequence;
+  }
+
+  /**
+   * Compute the weights of the candidates in the given list that have not been assigned a weight.
+   *
+   * @param candidates list of candidate sequences
+   * @return the total weight of all the elements in the candidate list
+   */
+  private double computeWeightForCandidates(SimpleList<Sequence> candidates) {
+    double totalWeight = 0.0;
+
+    // Iterate through the candidate list, computing the weight for a sequence only if it has
+    // not yet been computed before.
+    for (int i = 0; i < candidates.size(); i++) {
+      Sequence candidate = candidates.get(i);
+
+      Double weight = weightMap.get(candidate);
+      if (weight == null) {
+        weight = computeWeightForCandidate(candidate);
+      }
+
+      totalWeight += weight;
+    }
+
+    return totalWeight;
+  }
+
+  /**
+   * Compute the weight of a sequence. The formula for a sequence's weight is:
+   *
+   * <p>1.0 / (k * seq.exec_time * sqrt(seq.meth_size))
+   *
+   * <p>where k is the number of selections of seq and exec_time is the execution time of seq and
+   * meth_size is the number of method call statements in seq. This formula is a slight
+   * simplification of the one described in the GRT paper which maintains a separate exec_time for
+   * each execution of seq. However, we assume that every execution time for a sequence is the same
+   * as the first execution.
+   *
+   * @param sequence the sequence to compute a weight for
+   * @return the computed weight for the given sequence
+   */
+  private double computeWeightForCandidate(Sequence sequence) {
+    double methodSizeSqrt = getMethodSizeSquareRootForSequence(sequence);
+    Integer selectionCount = sequenceSelectionCount.get(sequence);
+    if (selectionCount == null) {
+      selectionCount = 1;
+    }
+
+    Long executionTime = sequenceExecutionTime.get(sequence);
+    if (executionTime == null) {
+      executionTime = 1L;
+    }
+
+    double weight = 1.0 / (selectionCount * executionTime * methodSizeSqrt);
+    weightMap.put(sequence, weight);
+
+    return weight;
+  }
+
+  /**
+   * To compute the execution time of an input sequence, we iterate through its statements and
```

Comment:
On this line, does "an input sequence" mean `eSeq`, or a member of `inputSequences`?
The name suggests `inputSequences`, but I would expect that their cost has already been computed.  `eSeq` makes more sense to me, but is in conflict with the phrase.  Could you please clarify?